### ISSUE TITLE
Document.cloneNode fails with external stylesheet links

### ISSUE THREAD

**Issue by chharvey** [jsdom/jsdom#2497]
### Basic info:

- **Node.js version:** v8.9.4
- **jsdom version:** v13.1.0

### Minimal reproduction case

```js
const jsdom = require('jsdom')

let doc = new jsdom.JSDOM(`
<!doctype html>
<html>
	<head>
		<link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700"/>
	</head>
	<body></body>
</html>
`).window.document

let clone = doc.cloneNode(true)
```

It seems to fail only with an external link. With an internal link (`<link href="./stylesheet.css"/>`), or if the `<link/>` is removed, it doesn’t throw the error.

### How does similar code behave in browsers?

`document.cloneNode(true)` works as expected in browsers.

### Stack Trace
```
Cannot read property 'fetch' of null
    at PerDocumentResourceLoader.fetch (/.../node_modules/jsdom/lib/jsdom/browser/resources/per-document-resource-loader.js:16:42)
    at fetchStylesheetInternal (/.../node_modules/jsdom/lib/jsdom/living/helpers/stylesheets.js:77:18)
    at exports.fetchStylesheet (/.../node_modules/jsdom/lib/jsdom/living/helpers/stylesheets.js:10:10)
    at obtainTheResource (/.../node_modules/jsdom/lib/jsdom/living/nodes/HTMLLinkElement-impl.js:82:3)
    at HTMLLinkElementImpl._attach (/.../node_modules/jsdom/lib/jsdom/living/nodes/HTMLLinkElement-impl.js:35:7)
    at HTMLHeadElementImpl._attach (/.../node_modules/jsdom/lib/jsdom/living/nodes/Node-impl.js:231:15)
    at HTMLHeadElementImpl._attach (/.../node_modules/jsdom/lib/jsdom/living/nodes/Element-impl.js:88:11)
    at HTMLHtmlElementImpl._attach (/.../node_modules/jsdom/lib/jsdom/living/nodes/Node-impl.js:231:15)
    at HTMLHtmlElementImpl._attach (/.../node_modules/jsdom/lib/jsdom/living/nodes/Element-impl.js:88:11)
    at DocumentImpl._insert (/.../node_modules/jsdom/lib/jsdom/living/nodes/Node-impl.js:685:14)
```

### Previous issues
- #833
- #967
- #2192

**Comment by domenic** [comment]
This appears to have been fixed as a result of ff5066f21a95b67df3f048305eddcb5854793257, but let's leave it open to track adding tests to prevent future regressions.

**Comment by chharvey** [comment]
(ff5066f is the merge commit for #2529)
